presto-hive/src/test/java/com/facebook/presto/hive/TestShowStats.java

```diff
@@ -106,6 +106,22 @@ public void testShowStats()
                         "   (null, null, null, null, 5.0, null, null))");
     }
 
+    @Test
+    public void testShowStatsNotSelecAllColumns()
+    {
+        assertQueryFails("SHOW STATS FOR (SELECT nationkey FROM nation_partitioned)", ".*Only SELECT \\* is supported in SHOW STATS SELECT clause");
+    }
+
+    @Test
+    public void testShowStatsWhereClause()
+    {
+        assertQueryFails("SHOW STATS FOR (SELECT nationkey FROM nation_partitioned WHERE nationkey > 0)", ".*Only columns that can be pushed down are supported in SHOW STATS WHERE clause");
+
+        assertQueryFails("SHOW STATS FOR (SELECT nationkey FROM nation_partitioned WHERE regionkey > 0 and nationkey > 0)", ".*Only columns that can be pushed down are supported in SHOW STATS WHERE clause");
+
+        assertQueryFails("SHOW STATS FOR (SELECT nationkey FROM nation_partitioned WHERE nationkey = 1 and name is not null)", ".*Only columns that can be pushed down are supported in SHOW STATS WHERE clause");
```

Comment:
what's this query testing that's different from the one above?

Reply:
This previous test case has one partition column and one non-partition column, this one has two non-partition columns.